test/e2e/storage/volumeattributesclass.go

```diff
@@ -171,4 +181,116 @@ var _ = utils.SIGDescribe("VolumeAttributesClass", feature.VolumeAttributesClass
 		}))
 		framework.ExpectNoError(err, "timeout while waiting to confirm VolumeAttributesClass %q deletion", updatedVolumeAttributesClass.Name)
 	})
+
+	framework.It("should be protected by vac-protection finalizer", func(ctx context.Context) {
+		vacClient := f.ClientSet.StorageV1beta1().VolumeAttributesClasses()
+
+		initialVAC := &storagev1beta1.VolumeAttributesClass{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "VolumeAttributesClass",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-",
+			},
+			DriverName: "e2e-fake-csi-driver",
+			Parameters: map[string]string{
+				"foo": "bar",
+			},
+		}
+
+		ginkgo.By("Creating a VolumeAttributesClass")
+		createdVAC, err := vacClient.Create(ctx, initialVAC, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "Failed to create the requested VolumeAttributesClass")
+		ginkgo.DeferCleanup(framework.IgnoreNotFound(vacClient.Delete), createdVAC.Name, metav1.DeleteOptions{})
+
+		ginkgo.By("Verifying that the vac-protection finalizer is set")
+		gomega.Eventually(getVolumeAttributesClassFinalizers(ctx, f.ClientSet, createdVAC.Name)).WithPolling(framework.Poll).WithTimeout(VacUpdatingTimeout).Should(
```

Comment:
suggestion changes:

```go
gomega.Expect(createdVAC.Finalizers).NotTo(gomega.ContainElement(volumeutil.VACProtectionFinalizer), "shouldn't contains the finalizer when the vac is created")
```

Reply:
Agree, but I use this previously but it seems a bit flaky in test records ->
https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/130553/pull-kubernetes-e2e-storage-kind-alpha-beta-features/1896869492995133440 not sure it is caused by sometimes the kind test cluster was a bit slow somehow. So I change to Eventually check.

Reply:
Don't remove your code, just add this check before your code. Once https://github.com/kubernetes/kubernetes/pull/130553#discussion_r1982655189 is merged, you can remove this check.

Reply:
Sorry for the confusion

Reply:
Never mind, good to know, thanks for the clarify!